src/app/src/test/kotlin/org/imperial/mrc/hint/integration/IntegrationTests.kt

```diff
@@ -76,6 +76,35 @@ abstract class SecureIntegrationTests : CleanDatabaseTests()
                 getTestEntity("survey.csv"))
     }
 
+    fun assertSecureWithSuccess(isAuthorized: IsAuthorized,
+                                responseEntity: ResponseEntity<String>,
+                                schemaName: String?, httpStatus: HttpStatus)
+    {
+
+        when (isAuthorized)
+        {
+            IsAuthorized.TRUE ->
+            {
+                Assertions.assertThat(responseEntity.headers.contentType!!.toString())
+                        .contains("application/json")
+
+                if (responseEntity.statusCode != httpStatus)
+                {
+                    Assertions.fail<String>("Expected BAD_REQUEST response but got error: ${responseEntity.body}")
```

Comment:
Since the function takes HttpStatus as an argument, here we want to pass in the expected status, not hard-code `BAD_REQUEST`.  The message should probably be a little different too, maybe 

`"Expected $httpStatus but got $responseEntity.status"`